Cpp_test/lib/ipc-pipe-send.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ipc-pipe-send.h"
+
+IPCPipeSend::~IPCPipeSend()
+{
+	close(this->pd);
+	unlink(this->info.method_name);
+}
+
+void IPCPipeSend::init()
+{
+	this->file_handler.setup_file(info.file_name, FileMode::READ);
+	// remove old pipe (if any)
+	remove(this->info.method_name);
```

Comment:
Did you think what happen if the receiver is already connected to the previous pipe?

Reply:
It means that old pipe cannot be removed, I will set if statement here to check remove old pipe

Reply:
Technically the old pipe will be unlinked (no process can make a new "connection" to it but it will still be there until every process closed the pipe. So the sender will create another pipe and connect to it, and the receiver will be connected to another pipe.

Reply:
As I said, you will have an issue if there is an old pipe and the receiver is connected to it before the sender. This is not mandatory to take care of this issue right now but keep it in mind for the story Hands on programming 2.